ai16z/eliza
# feat: add goat plugin

# Relates to:

https://github.com/ai16z/eliza/issues/735
https://github.com/ai16z/eliza/issues/644

# Risks

Low

# Background
## What does this PR do?
[Goat](https://ohmygoat.dev) is an open-source framework for connecting AI agents to any onchain app.

This PR demonstrates how to integrate Goat with Eliza, enabling AI agents to execute a wide range of onchain actions with minimal effort. Using Goat’s plugin system, developers can enable agents to:
- Mint NFTs on Zora
- Swap tokens on Uniswap
- Place bets on Polymarket
all with just a single line of code.

The framework supports multichain by either defining actions for a specific set of chains or dynamically selecting chains per action.

This concrete example simply enables transferring and checking the balance for USDC and ETH on the Base network.

## What kind of change is this?
Features (non-breaking change which adds functionality)

## Why are we doing this? Any context or related work?
Enable to easily add more onchain capabilities by leveraging Goat.

# Documentation changes needed?
My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?
1. Go to `plugin-goat`

## Detailed testing steps
1. Make sure you have an `EVM_PRIVATE_KEY`, `EVM_PUBLIC_KEY` and `EVM_PROVIDER_URL` for Base set
2. Add `plugin-goat` to the dependencies of the `package.json` from the `agent` dir
3. Import it in `agent/src/index.ts`, exchange it for the `evmPlugin`
4. Run the agent and prompt it with: What's the USDC balance of `0x18Bead774f927Af586F86F6d054C269416E163DD`
5. Should give you the balance of `0x18Bead774f927Af586F86F6d054C269416E163DD`

To test actions on Base Sepolia simply:
1. Go to `plugin-goat/src/index.ts` and change
```typescript
chain: {
    type: "evm",
    id: base.id,
}, 
```
to
```typescript
chain: {
    type: "evm",
    id: sepolia.id, // including the viem import
}, 
```
2. Go to the `plugin-goat/src/wallet.ts` file and do the same
3. Make sure `EVM_PROVIDER_URL` works for Sepolia
4. Run the agent!


## Discord username
aguspunk
